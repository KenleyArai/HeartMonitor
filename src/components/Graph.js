import React, { Component } from "react";
import { View, Text } from "react-native";
import PropTypes from "prop-types";

import { AreaChart } from "react-native-svg-charts";
import * as shape from "d3-shape";

export default class Graph extends Component<{}> {
  static propTypes = {
    heart_samples: PropTypes.arrayOf(
      PropTypes.shape({
        value: PropTypes.number,
        timestamp: PropTypes.string
      })
    )
  };
  static defaultProps = {
    heart_samples: [
      {
        value: -1,
        startDate: "2018-03-02T06:12:54.536-0800"
      }
    ]
  };

  render() {
    let { heart_samples } = { ...this.props };
    return (
      <AreaChart
        style={{ height: 200 }}
        data={heart_samples.map(sample => sample["value"])}
        contentInset={{ top: 30, bottom: 30 }}
        curve={shape.curveNatural}
        svg={{ fill: "rgba(134, 65, 244, 0.8)" }}
      />
    );
  }
}
