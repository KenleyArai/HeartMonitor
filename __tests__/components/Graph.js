import "react-native";
import React from "react";
import Graph from "../../src/components/Graph";

// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<Graph />).toJSON();
  expect(tree).toMatchSnapshot();
});
