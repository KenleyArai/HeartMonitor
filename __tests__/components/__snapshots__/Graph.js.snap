// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "height": 200,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSVGSvgView
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "flex": 1,
          },
          false,
          undefined,
        ]
      }
    >
      <RNSVGGroup
        fill={
          Array [
            0,
            0,
            0,
            0,
            1,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        propList={Array []}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={null}
      >
        <RNSVGLine
          fill={
            Array [
              0,
              0,
              0,
              0,
              1,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          propList={
            Array [
              "stroke",
              "strokeWidth",
            ]
          }
          stroke={
            Array [
              0,
              0,
              0,
              0,
              0.2,
            ]
          }
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          x1="0%"
          x2="100%"
          y1="-30"
          y2="-30"
        />
      </RNSVGGroup>
      <RNSVGPath
        d="M0,-30L0,-30Z"
        fill={
          Array [
            0,
            0.5254901960784314,
            0.2549019607843137,
            0.9568627450980393,
            0.8,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        propList={
          Array [
            "fill",
          ]
        }
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={null}
      />
    </RNSVGSvgView>
  </View>
</View>
`;
